src/symbols/lva.ts

```diff
@@ -55,36 +55,35 @@ export class LiveVariableAnalyzer {
   private computeSymbolsWithAvailableReads(block: CfgBlock): Set<ts.Symbol> {
     const availableReads = this.successorSymbolsWithAvailableReads(block);
     [...block.getElements()].reverse().forEach(element => {
-      const usage = this.symbols.getUsage(usageNode(element));
-      if (usage && !this.isUsedInNestedFunctions(usage.symbol)) {
-        if (usage.is(UsageFlag.WRITE)) {
-          if (availableReads.has(usage.symbol)) {
-            usage.dead = false;
-            availableReads.delete(usage.symbol);
-          } else {
-            usage.dead = true;
-          }
-        }
-        if (usage.is(UsageFlag.READ)) {
-          availableReads.add(usage.symbol);
-        }
+      if (isAssignment(element)) {
+        siftIdentifiers((element as ts.BinaryExpression).left).identifiers.forEach(identifier => {
+          this.trackUsage(this.symbols.getUsage(identifier), availableReads);
+        });
+      } else {
+        this.trackUsage(this.symbols.getUsage(element), availableReads);
```

Comment:
@vilchik-elena in fact we did, it was inside the `usageNode` method. Look at the code as it was a few commits before, like in f943a7f0f311399dc28b12a943d0b56051225a76